src/mqueryfront/src/StorageList.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+import ErrorBoundary from "./ErrorBoundary";
+
+class StorageRow extends Component {
+    getWatchedBadge() {
+        if (this.props.enabled) {
+            return (
+                <span
+                    class="badge badge-success"
+                    data-toggle="tooltip"
+                    title="This location is watched and indexed automatically"
+                >
+                    watched
+                </span>
+            );
+        }
+        return (
+            <span
+                class="badge badge-secondary"
+                data-toggle="tooltip"
+                title="This location is not indexed automatically"
+            >
+                disabled
+            </span>
+        );
+    }
+
+    render() {
+        return (
+            <tr>
+                <td>
+                    <code>{this.props.name}</code>
+                    {this.props.taints.map((taint) => (
+                        <span>
+                            {" "}
+                            <span
+                                class="badge badge-primary"
+                                data-toggle="tooltip"
+                                title={`New indexed files are tagged with "${taint}"`}
+                            >
+                                {taint}
+                            </span>
+                        </span>
+                    ))}
+                </td>
+                <td>
+                    <code>{this.props.path}</code>
+                </td>
+                <td>
+                    {new Date(this.props.lastUpdate).toLocaleDateString()}{" "}
+                    {this.getWatchedBadge()}
+                </td>
+                <td>
+                    <button
+                        type="button"
+                        class="btn btn-secondary btn-sm"
+                        data-toggle="tooltip"
+                        title="Reindex this dataset now"
+                    >
+                        Reindex
+                    </button>
+                    <button
+                        type="button"
+                        class="btn btn-danger btn-sm"
+                        data-toggle="tooltip"
+                        title="Stop watching the dataset (keep indexed files)"
+                    >
+                        Delete
+                    </button>
```

Comment:
Gather the buttons in `btn-toolbar` to create space between them

Reply:
`btn-toolbar` didn't give the effect that I've wanted, but I've added a literal whitespace between them - looks better now I think
![ss](https://user-images.githubusercontent.com/7026881/79231488-6b4a0b00-7e66-11ea-9e15-df77ae7c6e2e.png)